pkgs/applications/audio/roomeqwizard/default.nix

```diff
@@ -0,0 +1,91 @@
+{ writeTextFile, jdk8, runCommand, fetchurl, lib, makeWrapper, writers
+, common-updater-scripts, perl, nixfmt, nix, curl, coreutils, makeDesktopItem }:
+
+let
+  pname = "roomeqwizard";
+  version = "5.20.4";
+
+  responseFile = writeTextFile {
+    name = "response.varfile";
+    text = ''
+      createDesktopLinkAction$Boolean=false
+      executeLauncherAction$Boolean=false
+      mem$Integer=1
+      opengl$Boolean=false
+      sys.adminRights$Boolean=false
+      sys.installationDir=INSTALLDIR
+      sys.languageId=en
+      sys.programGroupDisabled$Boolean=true
+        '';
+  };
+
+  updateScript = writers.writeBash "${pname}-update.sh" ''
+    set -euo pipefail
+
+    PATH=${
+      lib.makeBinPath [ curl common-updater-scripts nixfmt nix coreutils perl ]
+    }
+    perlexpr='if (/current version.{1,10}v(\d+)\.(\d+)\.(\d+)/i) { print "$1.$2.$3"; break; }'
+
+    oldVersion="$(nix-instantiate --eval -E "with import ./. {}; lib.getVersion ${pname}" | tr -d '"')"
+    latestVersion="$(curl -sS https://www.roomeqwizard.com/index.html | perl -ne "$perlexpr")"
+
+    if [ ! "$oldVersion" = "$latestVersion" ]; then
+      update-source-version ${pname} "$latestVersion" --version-key=version --print-changes
+      nixfmt "pkgs/applications/audio/roomeqwizard/default.nix"
+    else
+      echo "${pname} is already up-to-date"
+    fi
+  '';
+
+in runCommand "roomeqwizard-${version}" {
+  inherit pname version;
+
+  src = fetchurl {
+    url = "https://www.roomeqwizard.com/installers/REW_linux_${
+        lib.replaceChars [ "." ] [ "_" ] version
+      }.sh";
+    sha256 = "0m2b5hwazy4vyjk51cmayys250rircs3c0v7bv5mn28h7hyq29s8";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    icon = pname;
+    desktopName = "REW";
+    genericName = "Software for audio measurements";
+    categories = "AudioVideo;";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  passthru.updateScript = updateScript;
+
+  meta = with lib; {
+    homepage = "https://www.roomeqwizard.com/";
+    license = licenses.unfree;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ zaninime ];
+    description = "Room Acoustics Software";
+    longDescription = ''
+      REW is free software for room acoustic measurement, loudspeaker
+      measurement and audio device measurement.
+    '';
+  };
+} ''
+  # set JDK path in installer
+  sed -E 's|^#\s*(INSTALL4J_JAVA_HOME_OVERRIDE=)|\1${jdk8}|' $src > installer
+  chmod +x installer
+
+  sed -e "s|INSTALLDIR|$out/share/roomeqwizard|" ${responseFile} > response.varfile
+
+  export HOME=$(pwd)
+
+  ./installer -q -varfile response.varfile
+
+  mkdir -p $out/bin $out/share/icons/hicolor/256x256/apps
+  makeWrapper $out/share/roomeqwizard/roomeqwizard $out/bin/roomeqwizard \
+    --set INSTALL4J_JAVA_HOME_OVERRIDE ${jdk8}
+
+  cp -r "$desktopItem/share/applications" $out/share/
+  cp $out/share/roomeqwizard/.install4j/s_*.png "$out/share/icons/hicolor/256x256/apps/${pname}.png"
```

Comment:
Please don't substitute pname.

Reply:
Why not? It is also used in the desktop item.

Reply:
Hmmm, might be okay to do that here but substituting every occurrence possible makes the code harder to read.